examples/dataflow-python-examples/streaming-examples/nlp-sentiment-analysis-pipeline/src/beam/pipeline.py

```diff
@@ -0,0 +1,413 @@
+
+    Documentation:
+      https://cloud.google.com/natural-language/docs/basics#sentiment_analysis
+
+    Languages Available for NLP API Sentiment Analysis: 
+      https://cloud.google.com/natural-language/docs/languages#sentiment_analysis 
+
+    Attributes:
+      language: Language of tweets.
+      client: NLP API client instance.
+    """
+    def __init__(self, client=None):
+        self.language = "en"
+        self.client = client
+
+    def setup(self): # NOTE: setup() is only supported with Dataflow Runner or Direct Runner (Apache Beam 2.22.0)
+        if self.client is None:
+            self.client = language.LanguageServiceClient()
+
+    @retry.Retry(predicate=retry.if_exception_type(exceptions.ResourceExhausted), initial=10.0, maximum=400.0, multiplier=2.0)
+    def process(self, tweet):
+        if self.client is None:
+            self.client = language.LanguageServiceClient()
+
+        (time_value, tweet_text) = tweet
+
+        document = {
+            "content": tweet_text,
+            "type": enums.Document.Type.PLAIN_TEXT,
+            "language": self.language
+        }
+
+        encoding_type = enums.EncodingType.UTF8
+        response = self.client.analyze_sentiment(document, encoding_type=encoding_type)
+
+        yield (time_value, response.document_sentiment.score)
+
+
+class CombineScores(beam.CombineFn):
+    """Combines sentiment scores per single date.
+
+    Reports total sum, total count, and average for all elements in an
+    individual window for an individual date. Total running average for
+    all windows can be calculated and updated in BigQuery.
+    """
+    def create_accumulator(self):
+        return (0.0, 0)
+
+    def add_input(self, score_values, input):
+        (sum, count) = score_values
+        return sum + input, count + 1
+
+    def merge_accumulators(self, accumulators):
+        sums, counts = zip(*accumulators)
+        return sum(sums), sum(counts)
+
+    def extract_output(self, score_values):
+        (sum, count) = score_values
+        return (sum, count, sum/count if count else float('NaN'))
+
+
+class DictFormat(beam.DoFn):
+    """Formats results into dictionary format and adds window timestamps.
+
+    WriteToBigQuery requires input data to be converted to dictionary format.
+    """
+    def process(self, tweet_tuple, window=beam.DoFn.WindowParam):
+        (date, data_tuple) = tweet_tuple
+        window_start = window.start.to_utc_datetime().strftime("%m-%d-%Y (%H:%M:%S)")
+        window_end = window.end.to_utc_datetime().strftime("%m-%d-%Y (%H:%M:%S)")
+
+        yield {
+            'tweet_date': date,
+            'window_start_utc': window_start,
+            'window_end_utc': window_end,
+            'sentiment_score_sum': data_tuple[0],
+            'num_tweets': data_tuple[1],
+            'average_sentiment_for_window': data_tuple[2]
+        }
+
+
+def run(argv=None, save_main_session=True):
+    parser = argparse.ArgumentParser()
```

Comment:
IMHO better to move arg parsing into it's own function.
Why? 
arg parsing has nothing to do w/ running the pipeline (e.g. happens only client side not on dataflow workers).
This is more clear if you have a function that parses args and a run command that just consumes those args.

This is just a suggestion.